xbmc/addons/AddonDll.h

```diff
@@ -54,19 +76,51 @@ namespace ADDON
     std::string m_parentLib;
 
   private:
+    /*!
+     * @brief Main addon creation call function
+     *
+     * This becomes called only one time before a addon instance becomes created.
+     * If another instance becomes requested is this Create no more used. To see
+     * like a "int main()" on exe.
+     *
+     * @param[in] firstKodiInstance The first instance who becomes used.
+     *                              In case addon supports only one instance
+     *                              and not multiple together can addon use
+     *                              only one complete class for everything.
+     *                              This is used then to interact on interface.
+     * @return The status of addon after the creation.
+     */
+    ADDON_STATUS Create(KODI_HANDLE firstKodiInstance);
+
     bool CheckAPIVersion(int type);
 
     DllAddon* m_pDll;
     bool m_initialized;
     bool LoadDll();
     bool m_needsavedsettings;
+    std::map<std::string, std::pair<ADDON_TYPE, KODI_HANDLE>> m_usedInstances;
 
     virtual ADDON_STATUS TransferSettings();
 
     static void AddOnStatusCallback(void *userData, const ADDON_STATUS status, const char* msg);
     static bool AddOnGetSetting(void *userData, const char *settingName, void *settingValue);
     static void AddOnOpenSettings(const char *url, bool bReload);
     static void AddOnOpenOwnSettings(void *userData, bool bReload);
+
+    /// addon to kodi basic callbacks below
+    //@{
+    AddonGlobalInterface m_interface;
```

Comment:
Is here https://github.com/AlwinEsch/kodi/blob/12213e445eb79ab69e8f291dc0b4d2e08bd234f9/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h#L191-L216 because of the size
 is it not direct visible.

I add a note here to know direct the source